src/App.tsx

```diff
@@ -1,26 +1,270 @@
+      );
+      inputRef.current?.focus();
+    } catch {
+      setError(Errors.UpdateTodoFailed);
+      setTodos(prev =>
+        prev.map(t => (t.id === todoId ? { ...t, loading: false } : t)),
+      );
+    }
+  };
+
+  /* Delete todo */
+  const deleteTodoItem = async (todoId: number) => {
+    try {
+      setTodos(prev =>
+        prev.map(t => (t.id === todoId ? { ...t, loading: true } : t)),
+      );
+      await deleteTodo(todoId);
+      setTodos(prev => prev.filter(t => t.id !== todoId));
+      inputRef.current?.focus();
+    } catch {
+      setError(Errors.DeleteTodoFailed);
+      setTodos(prev =>
+        prev.map(t => (t.id === todoId ? { ...t, loading: false } : t)),
+      );
+    }
+  };
+
+  /* Derived data */
+  const filteredTodos = todos.filter(todo => {
+    if (filter === 'active') {
+      return !todo.completed;
+    }
+
+    if (filter === 'completed') {
+      return todo.completed;
+    }
+
+    return true;
+  });
+
+  const allCompleted = todos.length > 0 && todos.every(t => t.completed);
+  // Рахуємо лише Todo, які не completed і не loading
+  const activeCount = todos.filter(t => !t.completed && !t.loading).length;
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <Header
+        allCompleted={allCompleted}
+        title={title}
+        setTitle={setTitle}
+        onAddTodo={addTodo}
+        loading={loading}
+        filter={filter}
+        setFilter={setFilter}
+        onToggleAll={async () => {
+          try {
+            const newCompleted = !allCompleted;
+
+            setTodos(prev => prev.map(t => ({ ...t, loading: true })));
+            await Promise.all(
+              todos.map(t => updateTodo({ ...t, completed: newCompleted })),
+            );
+            setTodos(prev =>
+              prev.map(t => ({
+                ...t,
+                completed: newCompleted,
+                loading: false,
+              })),
+            );
+          } catch {
+            setError(Errors.ToggleAllFailed);
+            setTodos(prev => prev.map(t => ({ ...t, loading: false })));
+          }
+        }}
+        inputRef={inputRef}
+      />
+
+      <Main
+        todos={todos}
+        filteredTodos={filteredTodos}
+        toggleTodo={toggleTodo}
+        deleteTodoItem={deleteTodoItem}
+        loading={loading}
+      />
+
+      {todos.length > 0 && (
+        <Footer
+          todos={todos}
+          activeCount={activeCount}
+          filter={filter}
+          setFilter={setFilter}
+          onDeleteCompleted={async () => {
```

Comment:
move this function to other functions
```suggestion
          onDeleteCompleted={onDeleteCompleted}
```